retrieval/discovery/kubernetes/discovery.go

```diff
@@ -292,18 +297,31 @@ func (kd *Discovery) updateNodesTargetGroup() *config.TargetGroup {
 
 	// Now let's loop through the nodes & add them to the target group with appropriate labels.
 	for nodeName, node := range kd.nodes {
-		nodeAddress, err := nodeHostIP(node)
+		defaultNodeAddress, nodeAddressMap, err := nodeAddresses(node)
 		if err != nil {
 			log.Debugf("Skipping node %s: %s", node.Name, err)
 			continue
 		}
 
-		address := fmt.Sprintf("%s:%d", nodeAddress.String(), kd.Conf.KubeletPort)
+		kubeletPort := int(node.Status.DaemonEndpoints.KubeletEndpoint.Port)
```

Comment:
From Kubernetes 1.1 yes I would assume it to be present. I'm not sure what versions we should be supporting but from what I've seen the majority of Kubernetes users seem to upgrade reasonably quickly to take advantage of new features so wouldn't expect too many installations on < 1.1 (no hard numbers on that though).